### ISSUE TITLE
Withdraw transaction not broadcasted

### ISSUE THREAD

**Issue by louneskmt** [lightninglabs/pool#149]
Hi there!

I was using Lightning Pool with LiT v0.3.0, and I tried to withdraw some funds from my pool account. I got the following error: `[pool] rpc error: code = Unknown desc = account must be in StateOpen to bemodified`.

Meanwhile, I had some troubles with the connection to my node (I actually proxy my node onion url to localhost:10009 with socat, and make LiT connect to this last), so I restarted my node.

Now Pool seems to have taken in account the withdraw: my account `outpoint` and `latest_txid` have been updated, as well as the `available_balance` and `value` fields. The thing is that the transaction doesn't appear in any explorers, nor in my own node's, after almost two hours.

What can I do?

Let me know if you need more details.

lnd version: v0.11.1-beta 
Lightning Terminal version: v0.3.0

**Comment by Roasbeef** [comment]
What state is the account actually in?

**Comment by louneskmt** [comment]
> What state is the account actually in?

It is in `PENDING_UPDATE` state

**Comment by Roasbeef** [comment]
That state means it's waiting on an account update to confirm (deposit, withdrawal, etc) On start up, do you see either `poold` or `lnd` try to rebroadcast that transaction?

**Comment by louneskmt** [comment]
Here are the lnd logs when I restart LiT (and thus poold):
```
lnd | 2020-11-09 20:26:15.289 [INF] NTFN: New block epoch subscription
lnd | 2020-11-09 20:26:18.526 [INF] NTFN: New block epoch subscription
lnd | 2020-11-09 20:26:21.821 [INF] NTFN: New block epoch subscription
lnd | 2020-11-09 20:26:22.564 [INF] NTFN: New confirmation subscription: conf_id=2, txid=3cb7bb036d22e7c7b9d8e5243590fea3b48da89de3cc4b8a37b3bb797fcb6cd0, num_confs=3 height_hint=656194
lnd | 2020-11-09 20:26:24.928 [INF] DISC: Broadcasting 80 new announcements in 8 sub batches
lnd | 2020-11-09 20:26:50.187 [INF] CRTR: Processed channels=0 updates=43 nodes=4 in last 59.999744164s
```
And LiT startup logs:
```
2020-11-09 21:26:14.030 [INF] RPCS: Starting trader server
2020-11-09 21:26:16.042 [INF] LOOP: Starting event loop at height 656194
2020-11-09 21:26:16.645 [INF] RPCS: Connected to lnd node 03273abe63c50f1c5fb0 with pubkey 03273abe63c50f1c5fb03bea4d0663e01f8c422ab0613e7e90cb53589ba26a36a4
2020-11-09 21:26:20.897 [INF] ACCT: Waiting for 3 confirmation(s) of account 027a25e43b99cd1b70d8a42be9e7a0af948280c58cbef0fbba6534238ddad35220
2020-11-09 21:26:24.180 [INF] RPCS: Trader server is now active
```

Where `027a25e43b99cd1b70d8a42be9e7a0af948280c58cbef0fbba6534238ddad35220` is my trader key and `3cb7bb036d22e7c7b9d8e5243590fea3b48da89de3cc4b8a37b3bb797fcb6cd0` the non broadcasted withdraw txid.

**Comment by wpaulino** [comment]
Do you have any logs at the time you performed the withdrawal that could point to the issue?

**Comment by louneskmt** [comment]
I found these LiT logs:
```
2020-11-09 15:55:12.079 [INF] LITD: Handling gRPC request: /poolrpc.Trader/ListOrders
2020-11-09 15:55:13.244 [INF] LITD: Handling gRPC request: /poolrpc.Trader/Leases
2020-11-09 15:56:51.815 [INF] LITD: Handling gRPC request: /poolrpc.Trader/ListAccounts
2020-11-09 15:58:17.989 [INF] LITD: Handling gRPC request: /poolrpc.Trader/WithdrawAccount
2020-11-09 15:59:20.120 [INF] LITD: Handling gRPC request: /poolrpc.Trader/WithdrawAccount
2020-11-09 16:01:24.727 [INF] LITD: Handling gRPC request: /poolrpc.Trader/CloseAccount
2020-11-09 16:01:44.352 [INF] LITD: Handling gRPC request: /poolrpc.Trader/ListOrders
2020-11-09 16:01:54.095 [INF] LITD: Handling gRPC request: /poolrpc.Trader/CancelOrder
2020-11-09 16:02:04.361 [INF] LITD: Handling gRPC request: /poolrpc.Trader/CloseAccount
2020-11-09 16:02:48.681 [INF] LITD: Handling gRPC request: /poolrpc.Trader/CloseAccount
2020-11-09 16:02:56.881 [INF] SGNL: Received interrupt
2020-11-09 16:02:56.881 [INF] SGNL: Shutting down...
2020-11-09 16:02:56.881 [INF] SGNL: Gracefully shutting down.
2020-11-09 16:02:56.882 [INF] LITD: Shutdown signal received
2020-11-09 16:02:56.882 [INF] LOOP: Swap client terminating
2020-11-09 16:02:56.882 [INF] LOOP: Swap client terminated
2020-11-09 16:02:56.884 [ERR] ACCT: Unable to determine spend for account 027a25e43b99cd1b70d8a42be9e7a0af948280c58cbef0fbba6534238ddad35220: rpc error: code = Unavailable desc = transport is closing
2020-11-09 16:02:56.884 [ERR] LNDC: Error closing client connection: rpc error: code = Canceled desc = grpc: the client connection is closing
2020-11-09 16:02:56.883 [ERR] ACCT: Unable to receive block notification: rpc error: code = Unavailable desc = transport is closing
2020-11-09 16:03:00.007 [INF] LITD: Dialing lnd gRPC server at 127.0.0.1:10009
2020-11-09 16:03:00.114 [INF] LITD: Listening for http_tls on: [::]:443
2020-11-09 16:03:05.009 [WRN] GRPC: grpc: addrConn.createTransport failed to connect to {127.0.0.1:10009  <nil> 0 <nil>}. Err: connection error: desc = "transport: authentication handshake failed: context deadline exceeded". Reconnecting...
2020-11-09 16:03:09.951 [INF] LITD: Handling gRPC request: /poolrpc.Trader/ListAccounts
2020-11-09 16:03:14.671 [INF] LITD: Dialing lnd gRPC server at 127.0.0.1:10009
2020-11-09 16:03:14.779 [INF] LITD: Listening for http_tls on: [::]:443
2020-11-09 16:03:19.675 [WRN] GRPC: grpc: addrConn.createTransport failed to connect to {127.0.0.1:10009  <nil> 0 <nil>}: didn't receive server preface in time. Reconnecting...
2020-11-09 16:03:22.876 [INF] LNDC: Creating lnd connection to 127.0.0.1:10009
2020-11-09 16:03:22.876 [INF] LNDC: Connected to lnd
2020-11-09 16:03:25.676 [WRN] GRPC: grpc: addrConn.createTransport failed to connect to {127.0.0.1:10009  <nil> 0 <nil>}: didn't receive server preface in time. Reconnecting...
2020-11-09 16:03:34.433 [INF] LNDC: lnd version: v0.11.1-beta, build tags 'autopilotrpc,signrpc,walletrpc,chainrpc,invoicesrpc,watchtowerrpc'
2020-11-09 16:03:34.435 [INF] LNDC: Using network mainnet
2020-11-09 16:03:34.435 [INF] LNDC: Waiting for lnd to be fully synced to its chain backend, this might take a while
2020-11-09 16:03:38.268 [INF] LNDC: lnd is now fully synced to its chain backend
2020-11-09 16:03:38.451 [INF] STORE: Checking for schema update: latest_version=4, db_version=4
2020-11-09 16:03:38.573 [INF] LOOP: Connected to lnd node '03273abe63c50f1c5fb0' with pubkey 03273abe63c50f1c5fb03bea4d0663e01f8c422ab0613e7e90cb53589ba26a36a4 (version v0.11.1-beta, build tags 'autopilotrpc,signrpc,walletrpc,chainrpc,invoicesrpc,watcht
owerrpc')
2020-11-09 16:03:38.574 [INF] LOOP: Wait for first block ntfn
2020-11-09 16:03:38.746 [INF] CLDB: Checking for schema update: latest_version=1, db_version=1
2020-11-09 16:03:41.369 [INF] LOOP: Starting event loop at height 656152
2020-11-09 16:03:46.888 [INF] ACCT: Waiting for 3 confirmation(s) of account 027a25e43b99cd1b70d8a42be9e7a0af948280c58cbef0fbba6534238ddad35220
```

At 15:58, I tried to withdraw, and I got the `[pool] rpc error: code = Unknown desc = account must be in StateOpen to be modified` error, so I tried a second time. As it was not working, I tried to close the whole account, so I canceled my pending order. None of my withdraw nor close commands worked, so I restarted LiT. And from there, I started to face some gRPC connection errors, as you can see.

Hope this can help.

**Comment by wpaulino** [comment]
There must've been a previous withdrawal/deposit attempt prior to the one at 15:58 for you to see that error message, do you have any details regarding that? What response did you get when trying to close the account?

**Comment by louneskmt** [comment]
There haven't been another withdraw/deposit attempt prior to the one at 15:58. Just a successful deposit. 

I checked account status just before trying the first withdraw, and its state was `OPEN` as you can see below.
<img width="1522" alt="image" src="https://user-images.githubusercontent.com/38882571/98603516-7670f680-22e2-11eb-9217-441bfe921e53.png">

All my close attempts resulted in this error: `[pool] rpc error: code = Unknown desc = no order found`.
```
./pool --rpcserver=localhost:443 accounts close --trader_key 027a25e43b99cd1b70d8a42be9e7a0af948280c58cbef0fbba6534238ddad35220 --addr 3HQV4TJ2oqQzM4htRJMAJL4bRnrFbyV1TE --sat_per_vbyte 10
```

**Comment by wpaulino** [comment]
Ah I see what happened. In your first attempt, everything went fine on your side, but requesting the signature from the auctioneer failed due to it thinking the account hadn't confirmed yet (minimum 3 confs). Was the first withdrawal attempt issued shortly after the successful deposit? On your second attempt, it failed on your side since there's a pending withdrawal.

**Comment by louneskmt** [comment]
The [deposit](https://mempool.space/tx/fcd7187092082b974d738afd69e9fb9b155709fb12246194c140137ce73902e6) has been made more than 24 hours before my withdraw attempt. The only action I did between the deposit and my withdraw attempt is opening an order (a couple of minutes after the confirmation of the deposit), excepting listing orders and leases.

**Comment by louneskmt** [comment]
@wpaulino I see that you created a pull request fixing future issues like this one. Will it solve mine as well, or do I have something else to do to recover these funds / broadcast the transaction?

**Comment by wpaulino** [comment]
Unfortunately the damage is already done to your account, we'll need to fix it manually. I'll get back to you with steps on how to recover your funds.

**Comment by louneskmt** [comment]
> Unfortunately the damage is already done to your account, we'll need to fix it manually. I'll get back to you with steps on how to recover your funds.

Any news? @wpaulino

**Comment by wpaulino** [comment]
We're working on fixing your account on our end, we should be able to provide you a patch in the next day or two. Could you send me a DM on Slack (`@wilmer`, same photo)?

**Comment by wpaulino** [comment]
Closing as the issue should be resolved going forward. A patch was provided to fix the local account state.
